pkg/pillar/cmd/volumemgr/updatestatus.go

```diff
@@ -227,6 +227,22 @@ func doUpdateContentTree(ctx *volumemgrContext, status *types.ContentTreeStatus)
 				blobErrorEntities = append(blobErrorEntities, &types.ErrorEntity{EntityID: blob.Sha256, EntityType: types.ErrorEntityContentBlob})
 
 				leftToProcess = true
+				if blob.IsErrorSource(types.VerifyImageStatus{}) && blob.RetryCount < blobDownloadMaxRetries {
+
+					// Remove VerifyImage config and retry download
+					MaybeRemoveVerifyImageConfig(ctx, blobSha)
+					retryDownload(ctx, blobSha)
+
+					// Increment retry count
+					blob.RetryCount++
+					blob.State = types.DOWNLOADING
+					// Remove VerifyImageConfig reference
+					blob.HasVerifierRef = false
+					blob.ClearErrorWithSource()
```

Comment:
Elsehwere we clear HasVerifierRef immediately after the call to MaybeRemvoeVerifyImageConfig and we also clear blob.Path. I think grouping them that way makes it more consistent and correct.

Reply:
I changed the function MaybeRemoveVerifyImageConfig and added blob.HasVerifierRef = false in it. I did something similar for MaybeRemoveDownloadConfig.

Reply:
Ok